integration-tests/bats/feature-version.bats

```diff
@@ -130,3 +130,17 @@ SQL
     dolt --feature-version $OLD checkout other
     popd
 }
+
+@test "feature-version: dolt checkout can be used when a working set has a different feature version than its head" {
+    rm -rf .dolt/
+    dolt --feature-version $OLD init
+    dolt --feature-version $NEW sql -q "create table a (b int);"
+    dolt --feature-version $NEW checkout -b other
+    run dolt --feature-version $NEW status
+    [ "$status" -eq 0 ]
+    [[ "$output" =~ "other" ]] || false
+    dolt --feature-version $NEW checkout main
```

Comment:
Should also have a test for the new branch case

Reply:
It does, on line 138.